### PR TITLE
HDDS-3556 Refactor conf in SCMRatisServer to Java-based conf.

### PR DESCRIPTION
**Pull request by timmylicheng**
## What changes were proposed in this pull request?
Refactor SCMRatisServer conf to java based.
(Please fill in changes proposed in this fix)

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-3556

(Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HDDS-XXXX. Fix a typo in YYY.)

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?
UT
(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 1]
```diff
@@ -1141,9 +1142,11 @@ private void initializeRatisServer() throws IOException {
     if (scmRatisServer == null) {
       SCMNodeDetails scmNodeDetails = SCMNodeDetails
           .initStandAlone(configuration);
-      //TODO enable Ratis ring
-      scmRatisServer = SCMRatisServer.newSCMRatisServer(configuration, this,
-          scmNodeDetails, Collections.EMPTY_LIST);
+      //TODO enable Ratis group
+      scmRatisServer = SCMRatisServer.newSCMRatisServer(configuration
+              .getObject(SCMRatisServer.SCMRatisServerConfiguration.class),
+          this, scmNodeDetails, Collections.EMPTY_LIST,
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:1148]: Passing half created object (this) to SCMRatisServer constructor could cause subtle bugs that are tricky to get right. We should avoid it if possible.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/ratis/SCMRatisServer.java [hunk 2]
```diff
@@ -479,4 +425,166 @@ private void setServerRole(RaftPeerRole currentRole,
       this.roleCheckLock.writeLock().unlock();
     }
   }
+
+  /**
+   * Configuration used by SCM Ratis Server.
+   */
+  @ConfigGroup(prefix = "ozone.scm.ratis")
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/ratis/SCMRatisServer.java:432]: Is it possible to make these RATIS related configuration reusable among DN,SCM and OM? 
Just different prefix like, ozone.om.ratis, ozone.scm.ratis, ozone.dn.ratis as prefix?

Comment (timmylicheng) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/ratis/SCMRatisServer.java:432]: Yea it looks like we currently have ozone.om.ratis, ozone.scm.ratis, hdds.datanode.ratis, dfs.container.ratis and even dfs.ratis.

If we decide to refactor all config into Java-based configuration, we shall consider refactoring all config to be like ozone.ratis.*

Then we can take a look at reusable configuration among OM, SCM and container. I create Jira to track: https://issues.apache.org/jira/browse/HDDS-3577

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java (hunk 1 above)
Comment (timmylicheng) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: I have made updates to have SCM use static method for createSCM and make constructor private. SCMRatisServer will be constructed after SCM is constructed.

**Comment by ChenSammi** [comment]
+1, LGTM.

**Pull request merged**